### ISSUE TITLE
Add Neo4j plugin for Liquibase dependency when Liquibase and Spring Data Neo4j are selected

### ISSUE THREAD

**Issue by fbiville** [spring-io/start.spring.io#716]
The [Neo4j plugin for Liquibase](https://github.com/liquibase/liquibase-neo4j/) has recently been released.
It would be nice to automatically add it to the dependency list when both Liquibase and (EDIT: Spring Data) Neo4j are selected.

**Comment by snicoll** [comment]
@fbiville can you show a concrete usage of this please?

**Comment by fbiville** [comment]
With Liquibase and the Neo4j plugin in the classpath, you can run Liquibase change sets with Cypher statements.
Here is a simple example of what it looks like:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<databaseChangeLog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">

    <changeSet id="my-movie-init" author="fbiville">
        <comment>Creates a forgettable movie</comment>
        <sql>CREATE (:Movie {title: 'My Life'})</sql>
    </changeSet>
</databaseChangeLog>
```

**Comment by snicoll** [comment]
Sorry but that's not what I meant by concrete usage. Spring Boot has no support for Liquibase and Neo4j so I am not sure what's supposed to happen concretely when the extra dependency is in place.

**Comment by fbiville** [comment]
You can select Liquibase and Spring Data Neo4j in start.spring.io (see screenshot below). When that happens, it could make sense to automatically add the Neo4j plugin for Liquibase.
 
![Screenshot 2021-06-17 at 10 37 43](https://user-images.githubusercontent.com/445792/122362008-0e5d4f80-cf58-11eb-8c90-f80c8627ebf9.png)

**Comment by snicoll** [comment]
We're not making great progress here I am afraid. 

>  Spring Boot has no support for Liquibase and Neo4j 

I am aware we can select Neo4j and Liquibase on start.spring.io but, as far as I know, that does not trigger any behavior that the user could benefit in Spring Boot. I am trying to understand what a generated project would look like and what would happen from an application standpoint.

To make hopefully more progress, can you please create a project from start.spring.io and add the necessary bits that showcase the use case you have in mind?

**Comment by fbiville** [comment]
Just created a quick example: https://github.com/fbiville/neo4j-liquibase-sdn. Does that help?
Note: I can probably get rid of the JDBC starter if I rely on `org.springframework.boot.autoconfigure.liquibase.LiquibaseDataSource` instead.

**Comment by fbiville** [comment]
In other words, the use case is to run Liquibase change sets against the Neo4j instance targeted by a Spring Data Neo4j application. That is only possible if the Neo4j plugin for Liquibase is added to the classpath.

**Comment by eddumelendez** [comment]
My 2 cents, the only thing I see so far is to add the `org.liquibase.ext:liquibase-neo4j` dependency in the spring-boot dependency management or if there is a BOM (Bill Of Materials) from liquibase that spring-boot can use. But, from start.spring.io don't see any further changes.

**Comment by fbiville** [comment]
There is no BOM as far as I know, but the Neo4j plugin is aligned with Liquibase core, starting from 4.3.5.
Every version available for core >= 4.3.5 will be available for the Neo4j plugin as well.

**Comment by snicoll** [comment]
Thanks for the sample, it really was necessary to understand the request.

I don't think we should do anything here unless Spring Boot has some sort of auto-configuration for Liquibase and Neo4j. Without the extra configuration that you've shared with the sample, the addition of the plugin has little value IMO and can be confusing as the user might expect us to do something extra. If there's some sort of support in Spring Boot (if applicable), we can obviously reconsider.
